tests/basic/storage_pool_test.go

```diff
@@ -2053,4 +2055,486 @@ var _ = Describe("{VolUpdateAddDrive}", func() {
+}
+
+func inResync(vol string) bool {
+	volDetails, err := Inst().V.InspectVolume(vol)
+	fmt.Println(volDetails)
+	if err != nil {
+		log.Error("not in Resync State")
+		return false
+	}
+	for _, v := range volDetails.RuntimeState {
+		log.InfoD("RuntimeState is in state %s", v.GetRuntimeState()["RuntimeState"])
+		if v.GetRuntimeState()["RuntimeState"] != "resync" {
+			return false
+		}
+	}
+	return true
+}
+
+func WaitTillVolumeInResync(vol string) bool {
+	now := time.Now()
+	targetTime := now.Add(20 * time.Minute)
+
+	for {
+		if now.After(targetTime) {
+			log.Error("Current time is greater than 1 hour from now")
+			return false
+		} else {
+			if inResync(vol) {
+				return true
+			}
+		}
+	}
+}
+
+var _ = Describe("{PoolResizeVolumesResync}", func() {
+	var testrailID = 51301
+	// Testrail Description : Try pool resize when lot of volumes are in resync state
+	// Testrail Corresponds : https://portworx.testrail.net/index.php?/cases/view/51301
+	var runID int
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("PoolResizeVolumesResync", "Validate Pool resize when lots of volumes are in resync state", nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+
+	var contexts []*scheduler.Context
+	var vol_ids []string
+	stepLog := "should get the existing storage node and expand the pool by resize-disk"
+
+	It(stepLog, func() {
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("snapcreateresizepool-%d", i))...)
+		}
+		ValidateApplications(contexts)
+
+		time.Sleep(5 * time.Second)
+		for _, each := range contexts {
+			Volumes, err := Inst().S.GetVolumes(each)
+			fmt.Printf("List of Volumes Present in the system %s \n", Volumes)
+			log.FailOnError(err, "Listing Volumes Failed ")
+
+			for _, vols := range Volumes {
+				log.InfoD("Validating Node ID %s", vols.ID)
+				vol_ids = append(vol_ids, vols.ID)
+			}
+
+			// Select Random Volumes for pool Expand
+			randomIndex := rand.Intn(len(vol_ids))
+			randomVolIDs := vol_ids[randomIndex]
+
+			// From each volume pick the random pool and restart pxdriver
+			poolUUIDs, err := GetPoolIDsFromVolName(randomVolIDs)
+			log.InfoD("List of pool IDs %v", poolUUIDs)
+			log.FailOnError(err, "Failed to check if Journal enabled")
+
+			// Select the random pools from UUIDs for PxDriver Restart
+			randomIndex = rand.Intn(len(poolUUIDs))
+			rebootPoolID := poolUUIDs[randomIndex]
+
+			// Rebooting Node
+			log.InfoD("Get the Node for Restart %v", rebootPoolID)
+			restartDriver, err := GetNodeWithGivenPoolID(rebootPoolID)
+			log.FailOnError(err, "Geting Node Driver for restart failed")
+
+			isjournal, err := isJournalEnabled()
+			log.FailOnError(err, "Failed to check if Journal enabled")
+
+			poolToBeResized, err := GetStoragePoolByUUID(rebootPoolID)
+			log.InfoD("Pool to be resized %v", poolToBeResized)
+			log.FailOnError(err, "Failed to get pool using UUID ")
```

Comment:
Make this error more verbose for debugging purposes
`log.FailOnError(err, "Failed to get pool using UUID [%s]", rebootPoolID)`